src/services.ts

```diff
@@ -30,33 +32,161 @@ export class ServiceCompletionProvider implements vscode.CompletionItemProvider
+    public async createCompletionItems(
+        service: ApiClass,
+        operator: string,
+        inheritMembers = true,
+    ): Promise<vscode.CompletionItem[]> {
+        let completionItems: vscode.CompletionItem[] = []
+
+        for (const member of service.Members) {
+            if (member.MemberType === "Property") {
+                const security = member.Security as ApiPropertySecurity
+                if (security.Read !== "None" && security.Write !== "None") {
+                    continue
+                }
+            } else if (member.Security !== "None") {
+                continue
+            }
+
+            if (operator === ":") {
+                if (member.MemberType === "Function") {
+                    const params = []
+
+                    for (const param of member.Parameters) {
+                        const paramText = `${param.Name}${param.Default ? "?" : ""}: ${param.Type ? param.Type.Name : "unknown"}${param.Default ? ` = ${param.Default}` : ""}`
+                        params.push(paramText)
+                    }
+
+                    const completionItem = new vscode.CompletionItem(
+                        member.Name,
+                        vscode.CompletionItemKind.Method,
+                    )
+
+                    completionItem.detail = `(function) ${service.Name}:${member.Name}(${params.join(", ")}): ${member.ReturnType ? member.ReturnType.Name : "unknown"}`
+                    completionItem.documentation = new vscode.MarkdownString(`[Developer Reference](https://developer.roblox.com/en-us/api-reference/function/${service.Name}/${member.Name})`)
+                    completionItem.insertText = new vscode.SnippetString(`${member.Name}($0)`)
+
+                    completionItems.push(completionItem)
+                }
+            } else if (operator === ".") {
+                switch (member.MemberType) {
+                    case "Callback": {
+                        const params = []
+
+                        for (const param of member.Parameters) {
+                            const paramText = `${param.Name}: ${param.Type ? param.Type.Name : "unknown"}`
+                            params.push(paramText)
+                        }
+
+                        const completionItem = new vscode.CompletionItem(
+                            member.Name,
+                            vscode.CompletionItemKind.Constructor,
+                        )
+                        completionItem.detail = `(callback) ${service.Name}.${member.Name} = function (${params.join(", ")})`
+                        completionItem.documentation = new vscode.MarkdownString(`[Developer Reference](https://developer.roblox.com/en-us/api-reference/callback/${service.Name}/${member.Name})`)
+
+                        completionItems.push(completionItem)
+                        break
+                    }
+                    case "Event": {
```

Comment:
What's with the brackets for case?

Reply:
The brackets creates a new scope within the braces, which is necessary due to the `const completionItem =` definitions used within each one